nexus/tests/integration_tests/volume_management.rs

```diff
@@ -1114,7 +1112,210 @@ async fn test_create_image_from_snapshot_delete(
+
+    let _base_disk: Disk = NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, &disks_url)
+            .body(Some(&base_disk))
+            .expect_status(Some(StatusCode::CREATED)),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .unwrap()
+    .parsed_body()
+    .unwrap();
+
+    // Issue snapshot request
+    let snapshots_url = format!("/v1/snapshots?project={}", PROJECT_NAME);
+
+    let snapshot: views::Snapshot = object_create(
+        client,
+        &snapshots_url,
+        &params::SnapshotCreate {
+            identity: IdentityMetadataCreateParams {
+                name: "a-snapshot".parse().unwrap(),
+                description: String::from("you are on the way to destruction"),
+            },
+            disk: base_disk_name.clone().into(),
+        },
+    )
+    .await;
+
+    // Create an image from the snapshot
+    let image_create_params = params::ImageCreate {
+        identity: IdentityMetadataCreateParams {
+            name: "debian-11".parse().unwrap(),
+            description: String::from(
+                "you have no chance to survive make your time",
+            ),
+        },
+        source: params::ImageSource::Snapshot { id: snapshot.identity.id },
+        os: "debian".parse().unwrap(),
+        version: "12".into(),
+    };
+
+    let _image: views::Image =
+        NexusRequest::objects_post(client, "/v1/images", &image_create_params)
+            .authn_as(AuthnMode::PrivilegedUser)
+            .execute()
+            .await
+            .unwrap()
+            .parsed_body()
+            .unwrap();
+
+    assert_eq!(order.len(), 3);
+    for item in order {
+        // Still some crucible resources
+        assert!(!disk_test.crucible_resources_deleted().await);
+
+        match item {
+            DeleteImageTestParam::Image => {
+                let image_url = "/v1/images/debian-11";
+                NexusRequest::object_delete(client, &image_url)
+                    .authn_as(AuthnMode::PrivilegedUser)
+                    .execute()
+                    .await
+                    .expect("failed to delete image");
+            }
+
+            DeleteImageTestParam::Disk => {
+                NexusRequest::object_delete(client, &get_disk_url("base-disk"))
+                    .authn_as(AuthnMode::PrivilegedUser)
+                    .execute()
+                    .await
+                    .expect("failed to delete disk");
+            }
+
+            DeleteImageTestParam::Snapshot => {
+                let snapshot_url = get_snapshot_url("a-snapshot");
+                NexusRequest::object_delete(client, &snapshot_url)
+                    .authn_as(AuthnMode::PrivilegedUser)
+                    .execute()
+                    .await
+                    .expect("failed to delete snapshot");
+            }
+        }
+    }
+
+    // Assert everything was cleaned up
+    assert!(disk_test.crucible_resources_deleted().await);
+}
+
```

Comment:
I'd find it helpful to see a comment here about why you're testing with different orders.

Reply:
+1, done in bef3de54e